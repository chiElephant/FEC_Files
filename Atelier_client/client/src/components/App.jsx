import React, { useState, useEffect } from 'react';
import ReviewsModule from './reviews_module/ReviewsModule.jsx'

function App(props) {
  return (
    <div>
      <h1> The RedBean Atelier App </h1>
      <ReviewsModule
        productId={71698}
        productName={'Bright Future Sunglasses'}/>
    </div>
  );
}

export default App;
