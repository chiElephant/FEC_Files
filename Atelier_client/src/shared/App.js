/* global Image */

import React, { useState, useEffect, Suspense } from 'react';
import { useParams } from 'react-router-dom';

import axios from 'axios';
import Loading from './components/Loading.jsx';
import QandAModule from './components/questions_answers_module/QandAModule.jsx';
import ReviewsModule from './components/reviews_module/ReviewsModule.jsx';
import RelatedProducts from './components/RelatedProducts.jsx';
import Overview from './components/overview_module/Overview.jsx';

function App() {
  const { id: paramId } = useParams();
  const [productData, setProductData] = useState(null);
  const [relatedProducts, setRelatedProducts] = useState({});
  // const [loading, setLoading] = useState(true);
  // const [error, setError] = useState(null);

  console.log('productData: ', productData);
  console.log('paramId: ', paramId);

  // if (loading) {
  //   return <Loading />;
  // }

  // if (error) {
  //   return <div>Error: {error.message}</div>;
  // }

  return (
    <div id="components">
      <Overview productData={productData} />
      <RelatedProducts relatedProducts={relatedProducts} />
      <QandAModule
        product_id={productData.id}
        product_name={productData.name}
        questions_answers={productData.questionsWithAnswers}
      />
      <ReviewsModule
        product_id={productData.id}
        product_name={productData.name}
        reviewsData={productData.reviews}
        metaData={productData.metaData}
      />
    </div>
  );
}

export default App;
