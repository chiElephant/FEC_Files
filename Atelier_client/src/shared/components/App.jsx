import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Topbar from './Topbar.jsx';
import routes from '../routes';

export default function App({ serverData = null }) {
  return (
    <>
      <Topbar />
      <Routes>
        {routes.map((route) => {
          const { path, component: Component } = route;
          return (
            <Route
              key={path}
              path={path}
              element={<Component data={serverData} />}
            />
          );
        })}
      </Routes>
    </>
  );
}
