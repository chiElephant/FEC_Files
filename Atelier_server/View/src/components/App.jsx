import axios from 'axios';
import React, { useState, useEffect } from 'react';

import { useParams } from 'react-router-dom';
import Topbar from './Topbar.jsx';
import Overview from './overview_module/Overview.jsx';
// import QandAModule from './questions_answers_module/QandAModule.jsx';
// import ReviewsModule from './reviews_module/ReviewsModule.jsx';
// import ProductLinks from './ProductLinks.jsx';

function App() {
  const { id } = useParams();
  const [currentProduct, setCurrentProduct] = useState(null);
  // const [products, setProducts] = useState([]);

  useEffect(() => {
    const product_id = id || 1;
    axios.get(`/products/${product_id}`).then((res) => {
      setCurrentProduct(res.data);
    });
  }, [id]);

  // useEffect(() => {
  //   const options = {
  //     method: 'get',
  //     url: `/products`,
  //   };

  //   axios(options)
  //     .then((response) => {
  //       const productList = response.data;
  //       setProducts([...productList]);
  //     })
  //     .catch((error) => {
  //       console.log(error);
  //     });
  // }, []);

  const widgets =
    currentProduct === null ? (
      ''
    ) : (
      <>
        <Overview product_id={currentProduct.id} product={currentProduct} />
        {/* <QandAModule product_id={productId} product_name={productName} />
        <ReviewsModule product_id={productId} product_name={productName} /> */}
      </>
    );

  return (
    <div>
      <Topbar />
      {widgets}
      {/* <ProductLinks products={products} /> */}
    </div>
  );
}

export default App;
