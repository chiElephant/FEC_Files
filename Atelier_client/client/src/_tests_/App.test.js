import renderer from 'react-test-renderer';
import App from '../components/App.jsx';

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});