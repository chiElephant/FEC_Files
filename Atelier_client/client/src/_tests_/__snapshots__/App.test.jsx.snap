// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
     The RedBean Atelier App 
  </h1>
  <div
    id="reviews-module"
  >
    <h2>
       Ratings and Reviews 
    </h2>
    <div
      id="ratings-breakdown"
    >
      <h3>
        Ratings Breakdown
      </h3>
      <div>
        Average Rating: 
      </div>
      <div>
        Total Reviews: 
      </div>
      <br />
      <div
        id="rec-percent"
      >
        % of reviewers recommend this product
      </div>
      <br />
      <div />
      <br />
      <div>
        <h3>
          Product Breakdown
        </h3>
        <div
          id="char-div"
        >
          <div />
        </div>
      </div>
    </div>
    <div
      id="reviews-list"
    >
      <h3>
        Reviews List
      </h3>
      <div>
        <label>
          Sort on: 
        </label>
        <select
          id="sort-type"
          onChange={[Function]}
        >
          <option
            value="relevant"
          >
            Relevant
          </option>
          <option
            value="newest"
          >
            Newest
          </option>
          <option
            value="helpful"
          >
            Helpful
          </option>
        </select>
      </div>
      <ul
        id="reviews"
      >
        <div
          id="review"
        >
          <h4>
            Review
          </h4>
          <div>
            Avgerage Rating: 
            0
          </div>
          <div>
            Date Reviewed: 
          </div>
          <div>
            User: 
          </div>
          <h4 />
          <div />
          <div>
            <div />
          </div>
          <div />
          <div />
          <div>
            Helpfulness: 
            0
          </div>
        </div>
      </ul>
    </div>
    <div
      id="main-buttons"
    >
      <button
        onClick={[Function]}
      >
        Add Review
      </button>
      <button
        disabled={true}
        onClick={[Function]}
      >
        More Reviews
      </button>
    </div>
  </div>
  <div>
    <h1>
      Questions & Answers
    </h1>
    <div>
      <form
        onSubmit={[Function]}
      >
        <input
          placeholder="Have a question? Search for answers..."
          type="text"
        />
        <button
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div>
      <div>
        <button>
           More Answered Questions 
        </button>
        <button
          onClick={[Function]}
        >
           Add Question + 
        </button>
      </div>
    </div>
  </div>
</div>
`;
